### ISSUE TITLE
Private message is not private

### ISSUE THREAD

**Issue by cryptogun** [nitely/Spirit#157]
Private message should be differ from private topic. In a private conversation, you should ask for the opposite sides for permission before allowing a new person to join the conversation. Or just disable the invite function for private message.

**Comment by nitely** [comment]
Private topics are private. The owners of a private topic can invite whoever they want. That's how it works and it's the intended behaviour.

> In a private conversation, you should ask for the opposite sides for permission before allowing a new person to join the conversation.

That sounds like a different feature. For example private groups, where a member ask to join in, then they are allowed to view all topics created by that group.

> Or just disable the invite function for private message.

What would be the purpose of this? *Nothing is really private* if everyone involve don't want to keep it private (i.e they can take screen-shots of conversations and so on). Disabling the invite function would be crippling the private topics feature for no good reason, IMHO.

**Comment by cryptogun** [comment]
Yes, the feature of private topic is great. But many user may think private message is like SMS and meant to be more private then private topic. e.g. a man send a private message(love letter/booty call etc.) to a woman, then she invites many user into this private message, that would be embarrassing.

**Comment by nitely** [comment]
What keeps the woman from taking a screen-shot and sharing it?

**Comment by cryptogun** [comment]
He can lie :smiley_cat:
Neither can SMS. He should bear it. But SMS is a bit better then private topic. If he needs the invitation feature, he can use private topic. If he needs the SMS feature, we should provide the improved private message functionality.

**Comment by nitely** [comment]
Well, yeah haha. I guess we could disable the private topic invites and add a "Private Group Topic" where the invites are enabled. But I'm still not convinced about it.

**Comment by alesdotio** [comment]
I think the current implementation is fine. The only thing I changed on my forum is I separated topic notifications and private message notifications.

**Comment by cryptogun** [comment]
I just finish copying `private` folder to `SMS` folder and modifying SMS for private messages today. I made lots of modifications and now it works. I'll post my site link here once it's finished and up in case someone wanna see the final result. And I'll make some part of it opensource once I'm free.
As for PM/SMS, Here are some of the features/modifications:
- SMSs are all in the same topic (that has and only has two of the participants). No need to open new topic hence a better communication and a better/clean layout.
- Disable join and leave functions in a SMS topic so that it works like a SMS service. Author of a SMS topic can't kick the recipient out otherwise the recipient can't access the SMS history. User can "Unnotify" or "flag" in case spamming happens.

Con: Since user can modify their comments, that means they can modify the SMS they've sent to your phone. Analogy.
